lib/inspec/dependencies/resolver.rb

```diff
@@ -82,5 +92,41 @@ def resolve(deps, top_level = true, seen_items = {}, path_string = "") # rubocop
       Inspec::Log.debug("Dependency traversal complete.") if top_level
       graph
     end
+
+    def fallback_to_archive_on_fetch_failure(dep)
```

Comment:
Method name like `can_fallback_to_archive?` might be more expressive for what this method does

Reply:
My suggestion there is wrong because this method makes some changes to the current state, it doesn't just return a boolean.

Reply:
I would disagree with "very narrow use case" because it's the main use case for enterprise-scale customers (wrapper profile in an air gapped environment).
Maybe delete the current description paragraph and give 1-2 sentences that state what the method does and what the return condition is.

Reply:
Hrm, as written, that does read as editorializing. I'll revise. I don't mean that it is uncommon or unimportant, just that it is a very specific situation - a very special set of conditions have to be met in order for this feature to be able to work.